src/main/java/Kate.java

```diff
@@ -0,0 +1,293 @@
+        while (true) {
+            Scanner in = new Scanner(System.in);
+            String userInput = in.nextLine();
+
+            if (userInput.toUpperCase().equals("BYE")) {
+                break;
+            } else if (userInput.toUpperCase().startsWith("TODO ")) {
+                addToDo(userInput);
+            } else if (userInput.toUpperCase().startsWith("DEADLINE ")) {
+                addDeadline(userInput);
+            } else if (userInput.toUpperCase().startsWith("EVENT ")) {
+                addEvent(userInput);
+            } else if (userInput.toUpperCase().equals("LIST")) {
+                printTasks();
+            } else if (userInput.toUpperCase().startsWith("DONE ")) {
+                indicateDone(userInput);
+            } else {
+                printHelpPage();
+            }
+        }
+    }
+
+    /**
+     * Marks task as done for a particular task number
+     *
+     * @param userInput Input provided by user to indicate task completion for a task
+     */
+    private static void indicateDone(String userInput) {
+        String[] inputArr = userInput.split(" ");
+        boolean isValid = intChecker(inputArr, tasks.size());
+
+        if (!isValid) {
+            return;
+        }
+
+        int taskNumber = Integer.parseInt(inputArr[1]);
+        int taskNumberIndex = taskNumber - 1;
+        Task curTask = tasks.get(taskNumberIndex);
+        curTask.setDone();
+
+        String doneMessage = TEXT_INDENTATION + "Nice! I've marked this task as done:\n"
+                + TEXT_INDENTATION + "  " + curTask.printTaskInfo() + "\n";
+        printMessage(doneMessage);
+    }
+
+    /**
+     * Adds a task that is an event
+     *
+     * @param userInput An event the user wants to add
+     */
+    private static void addEvent(String userInput) {
+        String taskInfo = userInput.substring(EVENT_LENGTH).strip();
+        boolean isEmptyField = emptyFieldChecker(taskInfo, " /at ");
+
+        if (isEmptyField) {
+            printMessage(FAILURE_MESSAGE_ADD_EVENT);
+            return;
+        }
+
+        String[] infoArr = taskInfo.split(" /at ");
+        String taskDescription = infoArr[0].strip();
+        String timeFrame = infoArr[1].strip();
+
+        tasks.add(new Event(taskDescription, timeFrame));
+        printAddedTask();
+    }
+
+    /**
+     * Adds a task that has a deadline
+     *
+     * @param userInput A task a user wants to add that has a deadline
+     */
+    private static void addDeadline(String userInput) {
+        String taskInfo = userInput.substring(DEADLINE_LENGTH).strip();
+        boolean isEmptyField = emptyFieldChecker(taskInfo, " /by ");
+
+        if (isEmptyField) {
+            printMessage(FAILURE_MESSAGE_ADD_DEADLINE);
+            return;
+        }
+
+        String[] infoArr = taskInfo.split(" /by ");
+        String taskDescription = infoArr[0].strip();
+        String deadline = infoArr[1].strip();
+
+        tasks.add(new Deadline(taskDescription, deadline));
```

Comment:
Perhaps can extract this to a method 'handleDeadline()'? For same level abstraction.